java/src/main/java/system/gameplay/roleplay/DiceRoller.java

```diff
@@ -5,26 +5,53 @@
 import java.util.List;
 import java.util.Random;
 
-public class DiceRoller {
+public final class DiceRoller {
 
-    private final Random random;
+    private static final Random random = new Random();
 
-    public DiceRoller() {
-        random = new Random();
+    public static List<Integer> rollAbilityScores() {
+        //TODO: (A) This line is here so the compiler won't complain, remove it when you add the functionality for this method
+        ArrayList<Integer> rollAbilityScoresList = new ArrayList<>();
+        int topThreeSixSidedDie = topThreeSixSidedDie();
+        for (int i = 1; i <= 6; i++) {
```

Comment:
This is kind of weird. I can't really see any reason not to just get the first score inside of the for loop, and initialize `i` as `0` instead of `1` when declaring your loop.

Reply:
Oh interesting note on the garbage collector, and good point about the inefficient coding.